src/pudl/transform/eia.py

```diff
@@ -312,45 +312,36 @@ def _compile_all_entity_records(
         # clean_dfs with the name 'annual'. We don't want to harvest
         # from our newly harvested tables.
         # if the df contains the desired columns the grab those columns
-        if "annual" not in table_name and set(base_cols).issubset(
-            transformed_df.columns
-        ):
-            logger.debug(f"        {table_name}...")
-            # create a copy of the df to muck with
-            df = transformed_df.copy()
-            # we know these columns must be in the dfs
-            cols = []
-            # check whether the columns are in the specific table
-            for column in static_cols + annual_cols:
-                if column in df.columns:
-                    cols.append(column)
-            df = df[(base_cols + cols)]
-            df = df.dropna(subset=id_cols)
-            # add a column with the table name so we know its origin
-            df["table"] = table_name
-            dfs.append(df)
-            # check if there are any columns that should be renamed and harvested
+        if "annual" not in table_name:
+            if set(base_cols).issubset(transformed_df.columns):
```

Comment:
Previously, the `if map_cols_dict and ...` logic was nested inside this `if "annual" not in table_name and set(base_cols).issubset...` statement. However, in the case of the utility entity, the base columns are `report_date` and `utility_id_eia`, which aren't in the ownership table because the columns have been renamed to `operator_utility_id_eia` and `owner_utility_id_eia`. This meant that the ownership table wasn't making it past the first if statement (because the base cols weren't present) and no records were being harvested for the utility entity. So instead, I had to make it so we first check if the normal base columns are present, and then also check if the map_cols_dict maps to base column names in a separate statement.

I could make this `if "annual" not in table_name and base_cols.issubset ...` and then `if "annual" not in table_name and map_cols_dict...`. Not sure what is more conventional.

Reply:
is the `if "annual"` even required anymore? i think it was when we were passing around and editing the same  dict of dfs but if i understand correctly what is currently happening, "annual" should never be in the table name bc that is only in the table names of the harvested annual entity tables which should no long ever show up in `clean_dfs`